### PR TITLE
Fix --throw-if-absent behavior

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: tests/io-reader-index-of-test.toit [hunk 1]
```diff
@@ -15,21 +15,34 @@ main:
...
+            else:
+              expect-equals -1 pos
+              expect-throw io.Reader.UNEXPECTED_END_OF_READER:
+                (reader chunks test-buffered).index-of 'X' --to=to --throw-if-absent
+
+reader chunks/List buffered/bool -> io.Reader:
+  result := TestReader chunks
+  // If requested, we make sure all the reader data is buffered
+  // by looking for a character that isn't there.
+  if buffered: expect-equals -1 (result.index-of ' ')
```
Comment (floitsch) [inline tests/io-reader-index-of-test.toit:47]:
```suggestion
  if buffered: result.buffer-all
```

File: lib/io/reader.toit [hunk 2]
```diff
@@ -302,6 +302,14 @@ abstract class Reader implements old-reader.Reader:
   Returns -1 if $throw-if-absent is false and the $byte is not in the remaining data.
   */
   index-of byte/int --to/int?=null --throw-if-absent/bool=false -> int:
+    absent := :
```
Comment (floitsch) [inline lib/io/reader.toit:305]: Maybe we should then just provide a method?
For example?
```
index-of byte/int --to/int?=null [--if-absent] -> int?:
```

But probably for a different PR.

**Review by floitsch — APPROVED** [review]
LGTM (without the debug printing).

File: tests/io-reader-index-of-test.toit (hunk 1 above)
Comment (kasperl) [reply tests/io-reader-index-of-test.toit]: Done.

**Pull request merged**